doc/CMakeLists.txt

```diff
@@ -134,6 +144,26 @@ add_custom_target(
 
 set_property(DIRECTORY APPEND PROPERTY CMAKE_CONFIGURE_DEPENDS ${GEN_DEVICETREE_REST_SCRIPT})
 
+#-------------------------------------------------------------------------------
+# API overview
+
+set(GEN_API_OVERVIEW_REST_SCRIPT ${CMAKE_CURRENT_LIST_DIR}/_scripts/gen_api_overview_rest.py)
+
+add_custom_target(
+  api_overview
+  COMMAND ${CMAKE_COMMAND} -E env
+  ${PYTHON_EXECUTABLE} ${GEN_API_OVERVIEW_REST_SCRIPT}
+    --doxygen-xml-output-dir ${DOXY_OUT}/xml
+    ${DOCS_SRC_DIR}/develop/api
+  VERBATIM
+  USES_TERMINAL
+  COMMENT "Generating API overview documentation..."
+)
+
+set_property(DIRECTORY APPEND PROPERTY CMAKE_CONFIGURE_DEPENDS ${GEN_API_OVERVIEW_REST_SCRIPT})
+
+add_dependencies(api_overview doxygen)
```

Comment:
-1 on reintroducing scriptery in docs build (it required significant effort to clean up the mess we had with scripts). Please look into how Sphinx extensions are used. You can also find useful examples in sdk-nrf repository.